ui/src/common/components/__tests__/DateRangeFilter.test.jsx

```diff
@@ -1,202 +1,183 @@
+    advanceTo(currentDate);
+
+    const screen = render(
+      <DateRangeFilter onChange={onChange} range="2019-11-21--2019-11-23" />
     );
-    expect(onChange).toHaveBeenCalledWith('2019-04-21--2019-05-28');
+
+    fireEvent.click(screen.getByTestId('end-date-picker'));
+    fireEvent.click(screen.getByTitle('2019-11-22'));
+
+    expect(onChange).toHaveBeenCalledWith('2019-11-21--2019-11-22');
   });
 
   it('calls onChange when end date selected and no range', () => {
     const onChange = jest.fn();
-    const wrapper = shallow(<DateRangeFilter onChange={onChange} />);
-    const currentDate = advanceTo(new Date('2019-05-28T13:31:00+00:00'));
-    const onStartDateChanged = wrapper
-      .find('[data-test-id="end-date-picker"]')
-      .prop('onChange');
-    onStartDateChanged(
-      moment(currentDate),
-      moment(currentDate).format(DATE_RANGE_FORMAT)
-    );
+    const currentDate = new Date('2019-05-28T13:31:00+00:00');
+    advanceTo(currentDate);
+
+    const screen = render(<DateRangeFilter onChange={onChange} range="" />);
+    fireEvent.click(screen.getByTestId('end-date-picker'));
+    fireEvent.click(screen.getByTitle('2019-05-28'));
+
     expect(onChange).toHaveBeenCalledWith('--2019-05-28');
   });
 
   it('calls onChange with cleared start date when start date is removed and there are range', () => {
     const onChange = jest.fn();
-    const wrapper = shallow(
+    const screen = render(
       <DateRangeFilter onChange={onChange} range="2019-04-21--2019-11-22" />
     );
-    const onStartDateChanged = wrapper
-      .find('[data-test-id="start-date-picker"]')
-      .prop('onChange');
-    onStartDateChanged(null, '');
-    expect(onChange).toHaveBeenCalledWith('--2019-11-22');
-  });
 
-  it('calls onChange with cleared start date when start date is removed and no range', () => {
-    const onChange = jest.fn();
-    const wrapper = shallow(<DateRangeFilter onChange={onChange} />);
-    const onStartDateChanged = wrapper
-      .find('[data-test-id="start-date-picker"]')
-      .prop('onChange');
-    onStartDateChanged(null, '');
-    expect(onChange).toHaveBeenCalledWith(undefined);
+    const closeIcon = screen.getAllByRole('img', { name: 'close-circle' })[0]
+      .parentElement;
+    fireEvent.mouseUp(closeIcon);
+
+    expect(onChange).toHaveBeenCalledWith('--2019-11-22');
   });
 
   it('calls onChange with cleared end date when end date is removed and there are range', () => {
     const onChange = jest.fn();
-    const wrapper = shallow(
+    const screen = render(
       <DateRangeFilter onChange={onChange} range="2019-04-21--2019-11-22" />
     );
-    const onStartDateChanged = wrapper
-      .find('[data-test-id="end-date-picker"]')
-      .prop('onChange');
-    onStartDateChanged(null, '');
-    expect(onChange).toHaveBeenCalledWith('2019-04-21--');
-  });
 
-  it('calls onChange with cleared end date when end date is removed and no range', () => {
```

Comment:
same here ?

Reply:
Yes, same here, I was not able to reproduce this test case manually, as a user, not by manipulating props in code, so I don't think we need to test this, plus rtl behaves like a user so I think it would require hacking to perform.